File: firmware/commands.go
//go:build tinygo && !modemonly

package main

import "strings"

// BotCommand — единый источник для меню Telegram, /help и описаний.
type BotCommand struct {
	Cmd  string // без слэша: "balance"
	Desc string // для setMyCommands (коротко)
	Help string // для /help (можно чуть подробнее)
}

// botCommands — править только здесь.
var botCommands = []BotCommand{
	{Cmd: "balance", Desc: "Баланс (*100#)", Help: "/balance — баланс (*100#)"},
	{Cmd: "missed", Desc: "Пропущенные звонки", Help: "/missed — список пропущенных звонков"},
	{Cmd: "sms", Desc: "SMS: /sms +79… текст", Help: "/sms +79001112233 текст — отправить SMS"},
	{Cmd: "ussd", Desc: "USSD: /ussd *код#", Help: "/ussd *100# — произвольный USSD"},
	{Cmd: "help", Desc: "Справка", Help: "/help — справка"},
}

func helpText() string {
	var b strings.Builder
	b.WriteString("Команды:\n")
	for _, c := range botCommands {
		b.WriteString(c.Help)
		b.WriteByte('\n')
	}
	b.WriteString("\nВходящие SMS и пропущенные звонки приходят сюда автоматически.\n")
	b.WriteString("На экране: Next=inbox, OK=открыть/назад.")
	return b.String()
}

func cmdMatch(text, name string) bool {
	if text == "/"+name {
		return true
	}
	return strings.HasPrefix(text, "/"+name+"@")
}

File: firmware/src/commands.cpp
#include "commands.h"

const BotCommand BOT_COMMANDS[] = {
    {"balance", "Баланс (*100#)", "/balance — баланс (*100#)"},
    {"missed", "Пропущенные звонки", "/missed — список пропущенных звонков"},
    {"sms", "SMS: /sms +79… текст", "/sms +79001112233 текст — отправить SMS"},
    {"ussd", "USSD: /ussd *код#", "/ussd *100# — произвольный USSD"},
    {"help", "Справка", "/help — справка"},
};

const int BOT_COMMANDS_N = sizeof(BOT_COMMANDS) / sizeof(BOT_COMMANDS[0]);

String helpText() {
  String b = "Команды:\n";
  for (int i = 0; i < BOT_COMMANDS_N; i++) {
    b += BOT_COMMANDS[i].help;
    b += '\n';
  }
  b += "\nВходящие SMS и пропущенные звонки приходят сюда автоматически.\n";
  b += "На экране: Next=inbox, OK=открыть/назад.";
  return b;
}

bool cmdMatch(const String &text, const char *name) {
  String a = String("/") + name;
  if (text == a) return true;
  String b = a + "@";
  return text.startsWith(b);
}

bool parseSMSCmd(const String &text, String *number, String *body) {
  if (!text.startsWith("/sms ")) return false;
  String rest = text.substring(5);
  rest.trim();
  int sp = rest.indexOf(' ');
  if (sp < 0) return false;
  *number = rest.substring(0, sp);
  *body = rest.substring(sp + 1);
  number->trim();
  body->trim();
  return number->length() > 0 && body->length() > 0;
}
